### PR TITLE
feat: Add HPA for ArgoCD server and repoServer components

### PR DESCRIPTION
**Pull request by gpaul**
This PR adds a horizontal pod autoscaling to the server and repoServer components. The HPA template is inspired by the official nginx controller-hpa: https://github.com/helm/charts/blob/master/stable/nginx-ingress/templates/controller-hpa.yaml

Fixes https://github.com/argoproj/argo-helm/issues/218

Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by jdeprin** [comment]
Readme needs updated with added key/values for autoscaling.

**Comment by jdeprin** [comment]
Why is this chart version 1.8.3 and not 1.8.0?

**Comment by gpaul** [comment]
> Readme needs updated with added key/values for autoscaling.

Done.

> Why is this chart version 1.8.3 and not 1.8.0?

Fixed.

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -101,6 +101,11 @@ Helm v3 has removed the `install-crds` hook so CRDs are now populated by files i
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | repoServer.affinity | Assign custom affinity rules to the deployment https://kubernetes.io/docs/concepts/configuration/assign-pod-node/ | `{}` |
+| repoServer.autoscaling.enabled | Enable Horizontal Pod Autoscaler (HPA) for the repo server | `false` |
+| repoServer.autoscaling.minReplicas | Minimum number of replicas for the repo server HPA | `1` |
+| repoServer.autoscaling.maxReplicas | Minimum number of replicas for the repo server HPA | `5` |
```
Comment (cabrinha) [inline charts/argo-cd/README.md]: This should be "Maximum number of replicas for the repo server HPA"

File: charts/argo-cd/README.md [hunk 2]
```diff
@@ -143,6 +148,11 @@ Helm v3 has removed the `install-crds` hook so CRDs are now populated by files i
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | server.affinity | Assign custom affinity rules to the deployment https://kubernetes.io/docs/concepts/configuration/assign-pod-node/ | `{}` |
+| server.autoscaling.enabled | Enable Horizontal Pod Autoscaler (HPA) for the server | `false` |
+| server.autoscaling.minReplicas | Minimum number of replicas for the server HPA | `1` |
+| server.autoscaling.maxReplicas | Minimum number of replicas for the server HPA | `5` |
```
Comment (cabrinha) [inline charts/argo-cd/README.md]: Same here, "Maximum".

**Comment by cabrinha** [comment]
@gpaul do you think it would be a good idea to add HPA to the redis-ha subchart as well?

**Comment by seanson** [comment]
> @gpaul do you think it would be a good idea to add HPA to the redis-ha subchart as well?

That's a good question to raise but would call it outside the scope of this PR as it's non-trivial.

**Pull request merged**